tests/core/snapd-failover/task.yaml

```diff
@@ -1,64 +1,163 @@
+
+    if is_core16_system; then
+        echo "ensuring we reverted fully to core snap system"
+        not test -d /snap/snapd
     fi
-    cp -a local.conf.bak /etc/systemd/system/snapd.service.d/local.conf
-    systemctl daemon-reload
 
 execute: |
-    echo "Testing failover handling of the snapd snap"
-    current=$(readlink /snap/snapd/current)
-    SNAPD_SNAP=$(ls /var/lib/snapd/snaps/snapd_"$current".snap)
-
-    # for debugging
-    snap list --all
-
-    echo "Verify that a random signal does not trigger the failure handling"
-    echo "and snapd is just restarted"
-    systemctl kill --signal=SIGSEGV snapd.service
-    systemctl is-active snapd.failure.service | MATCH inactive
-    echo "Snap list is working still"
-    snap list | MATCH "^snapd .* $current .*"
-
-    echo "Break snapd"
-    unsquashfs -d ./snapd-broken "$SNAPD_SNAP"
-    echo "" > ./snapd-broken/usr/lib/snapd/snapd
-    (cd ./snapd-broken && snap pack .)
-    echo "Now install the broken snapd"
-    if snap install --dangerous ./snapd-broken/snapd_*.snap; then
-        echo "installing a broken snapd should not work, test broken"
-        exit 1
-    fi
+    # shellcheck source=tests/lib/journalctl.sh
+    . "$TESTSLIB/journalctl.sh"
+
+    # shellcheck source=tests/lib/systems.sh
+    . "$TESTSLIB/systems.sh"
+
+    if [ "$SPREAD_REBOOT" = 0 ]; then
+        echo "Testing failover handling of the snapd snap"
+
+        # for debugging
+        snap list --all
+
+        # get the number of times that snapd.failure was started before we do 
+        # anything
+        started_before="$(get_journalctl_log -u snapd.failure | grep -c 'Started Failure handling of the snapd snap.' || true)"
 
-    echo "And verify that snap commands still work and snapd is reverted"
-    retry-tool -n 30 --wait 5 systemctl is-active snapd.failure.service
-    snap list | MATCH "^snapd .* $current .*"
+        # test that random signals don't trigger snapd-failure
+        current=$(readlink /snap/snapd/current)
 
-    echo "Verify we got the expected error message"
-    snap change --last=install|MATCH "there was a snapd rollback across the restart"
+        echo "Verify that a random signal does not trigger the failure handling"
+        echo "and snapd is just restarted"
+        systemctl kill --signal=SIGSEGV snapd.service
 
-    echo "restart snapd and ensure we can still talk to it"
-    systemctl restart snapd.socket snapd.service
-    snap list | MATCH "^snapd .* $current .*"
+        started_after_rand_sig="$(get_journalctl_log -u snapd.failure | grep -c 'Started Failure handling of the snapd snap.' || true)"
+
+        if [ "$started_after_rand_sig" -gt "$started_before" ] ; then
+            echo "snapd.failure inadvertently started with a random signal!"
+            exit 1
+        fi
+
+        echo "Snap list is working still"
+        snap list | MATCH "^snapd .* $current .*"
+
+        SNAPD_SNAP=$(ls /var/lib/snapd/snaps/snapd_"$current".snap)
```

Comment:
Same comment as I had elsewhere. Unless current is a glob this feels like something that should be just a direct assignment.

Reply:
sure